core/src/util/usFramework.cpp

```diff
@@ -72,7 +73,7 @@ void Framework::Start()
       d->operation = BundlePrivate::OP_ACTIVATING;
       break;
     case STATE_ACTIVE:
-      return;
+      break;
```

Comment:
If I remember correctly, yes it was intentional. What wasn't intentional was the side-effect of the double activation.

The Framework does need to send the ```FRAMEWORK_STARTED``` event even if ```Framework::Start()``` is called multiple times. At least this is how I interpret the OSGi spec.

The ```break;``` allowed execution to continue and eventually send the ```FRAMEWORK_STARTED``` event.

I agree that double activation shouldn't happen. I'll fix this.